### DISCUSSION TITLE
High CPU Usage from Leftover Chrome Process

### DISCUSSION THREAD

**Discussion by nuweq** [seleniumbase/SeleniumBase#2861] [Q&A] [answered]
Does anyone know why a Chrome process remains active even after closing the browser and exiting ChromeDriver? This lingering process continues to use CPU resources, leading to high CPU usage and potentially slowing down the system, especially after running multiple tests. This issue occurs in both headless and non-headless modes. How can I resolve this issue? Thanks.

Chromedriver version = 126.0.6478.62
Seleniumbase version = 4.27.5
Operating system = Windows 11 Enterprise

![image](https://github.com/seleniumbase/SeleniumBase/assets/37559617/a7a2ae40-932e-4f73-89ac-fa91394c60d0)

**Comment by mdmintz** [comment, marked as answer]
See https://github.com/seleniumbase/SeleniumBase/issues/2505#issuecomment-1950096805

Only happens in UC Mode.

Also see https://github.com/seleniumbase/SeleniumBase/issues/2300#issuecomment-1822049273

**Reply by hkattab** [reply]
Try to introduce a timeout between driver requests, I had the same problem for a while when running the process as a task within the scheduler, so far it is working fine. I added a minimum 5000ms delay to help ensure that no overlap occurs.
FYI: the randomness is just to mimic human behavior!

randomValue = (CInt(Math.Floor((15 - 1 + 1) * Rnd())) + 1) * 1000 + 5000
Threading.Thread.Sleep(randomValue)
driver.Navigate.GoToUrl(url)
